src/miner.cpp

```diff
@@ -328,9 +334,115 @@ int BlockAssembler::UpdatePackagesForAdded(const CTxMemPool::setEntries& already
     return nDescendantsUpdated;
 }
 
-CTransaction BlockAssembler::CreateSidechainWTJoinTx(const uint8_t& nSidechain)
+CTransaction BlockAssembler::CreateSidechainWTJoinTx(uint8_t nSidechain)
```

Comment:
Agreed, what do you think about just making it return bool and CTransaction by reference?